core/trino-main/src/main/java/io/trino/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -1302,5 +1425,74 @@ public PlanNode getRoot()
         {
             return mappings;
         }
+
+        public Map<DynamicFilterId, DynamicFilterId> getDynamicFilterIdMap()
+        {
+            return dynamicFilterIdMap;
+        }
+
+        public PlanAndMappings with(PlanNode root, Map<Symbol, Symbol> mappings)
+        {
+            return new PlanAndMappings(root, mappings, dynamicFilterIdMap);
+        }
+    }
+
+    private static class DynamicFilterVisitor
+            extends PlanVisitor<PlanNode, UnaliasDynamicFilterContext>
```

Comment:
The `UnaliasDynamicFilterContext` is never modified while visiting the plan. It is effectively a constant. Consider keeping the map as private field of the Visitor.